src/main/webapp/js/map-loader.js

```diff
@@ -1,38 +1,133 @@
+let editMarker;
+let map;
+let displayMarkers = {};
  /**
  * Adds a new marker with a toggable info window
- * @param {Map} map 
  * @param {Number} lat 
  * @param {Number} lng 
- * @param {String} title 
  * @param {String} description 
  */
-function addLandmark(map, lat, lng, title, description){
+function addDisplayMarker(lat, lng, description){
   const marker = new google.maps.Marker({
     position: { lat: lat, lng: lng },
-    map: map,
-    title: title
+    map: map
   })
   const infoWindow = new google.maps.InfoWindow({
-    content: description
+    content: buildDeletableMarker( lat, lng, description)
   })
   marker.addListener('click', () => {
     infoWindow.open(map, marker)
   })
+
+  // Uses a dictionary with both lat and lng for faster lookup
+  if(!displayMarkers[lat]) displayMarkers[lat] = {};
+  displayMarkers[lat][lng] = marker;
 }
 
 /**
  * Main function
  */
 function createMap(){
-  const map = new google.maps.Map(document.getElementById('map'), {
+  map = new google.maps.Map(document.getElementById('map'), {
     center: {lat: 37.764419, lng: -122.445911},
     zoom: 12
   });
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  })
 
-  addLandmark(map, 37.819021, -122.477946, "Golden Gate Bridge", "This is the Golden Gate Bridge")
-  addLandmark(map, 37.802122, -122.418805, "Lombard Street", "This is Lombard St.")
-  addLandmark(map, 37.811039, -122.415175, "Fishermans Wharf", "This is Fishermans Wharf")
+  fetchMarkers();
+}
 
+ /** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+  fetch('/markers').then((response) => {
+    return response.json();
+  }).then((markers) => {
+    markers.forEach((marker) => {
+     addDisplayMarker(marker.lat, marker.lng, marker.content)
+    });
+  });
+}
 
+/** Creates a marker that shows a textbox the user can edit. */
+function createMarkerForEdit(lat, lng){
+  // If we're already showing an editable marker, then remove it.
+  if(editMarker){
```

Comment:
This is unnecessary, since you are going to overwrite editMarker entirely in line 59 anyways

Reply:
It works so that if you had previously clicked the map, the marker that was added will be removed from the map. The first time you click it, this will not run since editMarker will be undefined.

Reply:
I'm referring to:
```
 if(editMarker){
     editMarker.setMap(null);
  }
```
This is unnecessary.

Reply:
This may be an unexpected behavior. It does overwrite the variable, but on the actual map, the previous marker would not disappear.

The use case is:

- You click the map, a marker with a text box appears.
- You click on another part of the map. if ```.setMap(null)``` is not called, the first marker will stay there until you reload the page completely. The condition is just so it is not called on an undefined object when clicking the map for the first time.

Reply:
This is what happens without those 3 lines
![image](https://user-images.githubusercontent.com/31520553/60044654-7427e100-9688-11e9-85a1-1cef4352d743.png)